wordpress-mobile/gutenberg-mobile
# Issue/1966 Disable whitespace collapse in Aztec

Fixes #1966 

Related PRs:
 - Aztec iOS: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1277
 - Aztec Android: https://github.com/wordpress-mobile/AztecEditor-Android/pull/905
 - GB: https://github.com/WordPress/gutenberg/pull/21469

This PR fixes a bug where white space inserted at the start of text blocks ( Paragraph, Heading, Lists) were removed when applying a formatting option ( Bold, Italic, Strike).

The underlying issue was a sync issue between the content representation in the JS side and the Aztec content. 

Aztec by default removes extra spaces (like most browsers do), but GB internal format doesn't do that, so when spaces were sent to Aztec they were removed and then when we apply formats Aztec will send the content back with all the extra space removed.

This PR links a new version of Aztec that allows disabling the removal/collapse of extra spaces. 

To test:
 - Start the demo app
 - Add a text block, paragraph for example
 - Type some spaces, then write some text
 - Select the text and apply one format, for example, bold
 - Check that the white space aren't removed.
 - Repeat the test but without adding any text, just spaces, and then apply bold, to see if spaces are kept
 - Try to merge two paragraphs that have extra space at the start of the paragraph and on the end.
 - Check on the other blocks like Heading and Lists.

PR submission checklist:
